edx_django_utils/monitoring/code_owner/utils.py

```diff
@@ -61,48 +67,47 @@ def _process_code_owner_mappings():
         }
 
     """
-    _CODE_OWNER_MAPPINGS = getattr(settings, 'CODE_OWNER_MAPPINGS', None)
-    if not _CODE_OWNER_MAPPINGS:
+    global _PATH_TO_CODE_OWNER_MAPPINGS
+
+    # Return cached processed mappings if already processed
+    if _PATH_TO_CODE_OWNER_MAPPINGS is not None:
+        return _PATH_TO_CODE_OWNER_MAPPINGS
```

Comment:
I think there are thread-safety issues here that could be exposed on startup, as the first few requests race to complete this mapping. Maybe put a mutex around the method?

Reply:
(Everything else looks good.)

Reply:
Silly me.  I looked up how to do this in Django, and not Python, and didn't see anything simple so I did a hacked solution.  I can update with a mutex to replace the need for a comment.

Reply:
That looks good! There's one other case that occurs to me, which is that the global could be cleared after the conditional check here but before the return, so that a couple requests return None instead of a dict. But that clearing should only happen during testing, so I'm not worried. :-)

Reply:
Ok.  Leaving as-is.  Fast and cheap and won't lock.  Thanks.